### PR TITLE
[Fix] Make `IncludeDY` optional

### PR DESCRIPTION
**Pull request by IzaakWN**
### PR description

Make `IncludeDY` optional to fix for issues reported for PR #38829, e.g. https://github.com/cms-sw/cmssw/pull/38829#issuecomment-1217335165

### PR validation
Checked compilation with `scram`, code checks, code formats, and ran `cmsRun`.

### Backports
Pushed the same fixes to the backports https://github.com/cms-sw/cmssw/pull/38936 and https://github.com/cms-sw/cmssw/pull/38938.

### CODE REVIEW

**Comment by perrotta** [comment]
test parameters:
- workflow = 136.9
- addpkg = TauAnalysis/MCEmbeddingTools

**Comment by perrotta** [comment]
please test

File: TauAnalysis/MCEmbeddingTools/python/EmbeddingPythia8Hadronizer_cfi.py [hunk 1]
```diff
@@ -19,7 +19,8 @@
             MuHadCut = cms.string('Mu.Pt > 18 && Had.Pt > 25 && Mu.Eta < 2.1'),
             MuMuCut = cms.string('Mu1.Pt > 17 && Mu2.Pt > 8'),
             Final_States = cms.vstring('ElEl','ElHad','ElMu','HadHad','MuHad','MuMu'),
-            BosonPDGID = cms.int32(23)
+            BosonPDGID = cms.int32(23),
+            IncludeDY = cms.untracked.bool(True)
```
Comment (perrotta) [inline TauAnalysis/MCEmbeddingTools/python/EmbeddingPythia8Hadronizer_cfi.py:23]: Are you sure that you want to set `True` as default value?

Comment (IzaakWN) [reply TauAnalysis/MCEmbeddingTools/python/EmbeddingPythia8Hadronizer_cfi.py:23]: Ah yes, you are right. I was testing it `True` as well and forgot to update.

**Comment by perrotta** [comment]
please abort

**Comment by perrotta** [comment]
@cmsbuild please test

File: GeneratorInterface/Core/src/EmbeddingHepMCFilter.cc [hunk 2]
```diff
@@ -4,7 +4,8 @@
 #include "boost/algorithm/string/trim_all.hpp"
 
 EmbeddingHepMCFilter::EmbeddingHepMCFilter(const edm::ParameterSet &iConfig)
-    : ZPDGID_(iConfig.getParameter<int>("BosonPDGID")), includeDY_(iConfig.getParameter<bool>("IncludeDY")) {
+    : ZPDGID_(iConfig.getParameter<int>("BosonPDGID")),
+      includeDY_(iConfig.getUntrackedParameter<bool>("IncludeDY", false)) {
```
Comment (perrotta) [inline GeneratorInterface/Core/src/EmbeddingHepMCFilter.cc:8]: And, by the way (sorry if I am slow....): why untracked? It affects the output as the other config parameters do, and it should be tracked then?

Comment (IzaakWN) [reply GeneratorInterface/Core/src/EmbeddingHepMCFilter.cc:8]: How do I correctly make it a tracked, but optional parameter with a default value?

Comment (IzaakWN) [reply GeneratorInterface/Core/src/EmbeddingHepMCFilter.cc:8]: Just like this?
```
EmbeddingHepMCFilter::EmbeddingHepMCFilter(const edm::ParameterSet &iConfig)
  : ZPDGID_(iConfig.getParameter<int>("BosonPDGID")) {
    if (ps.exists("includeDY")) {
          includeDY_ = iConfig.getParameter<std::string>("IncludeDY");
    }
```
The default of `includeDY_` is already set to `false` in the header file.

File: TauAnalysis/MCEmbeddingTools/python/EmbeddingPythia8Hadronizer_cfi.py [hunk 3]
```diff
@@ -19,7 +19,8 @@
             MuHadCut = cms.string('Mu.Pt > 18 && Had.Pt > 25 && Mu.Eta < 2.1'),
             MuMuCut = cms.string('Mu1.Pt > 17 && Mu2.Pt > 8'),
             Final_States = cms.vstring('ElEl','ElHad','ElMu','HadHad','MuHad','MuMu'),
-            BosonPDGID = cms.int32(23)
+            BosonPDGID = cms.int32(23),
+            IncludeDY = cms.untracked.bool(False)
```
Comment (perrotta) [inline TauAnalysis/MCEmbeddingTools/python/EmbeddingPythia8Hadronizer_cfi.py:23]:
```suggestion
            IncludeDY = cms.bool(False)
```

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- The previosly failing workflow and unit tests now succed
- Let have this merged to remove the errors from the IBs
- @cms-sw/generators-l2 @cms-sw/simulation-l2 : the fix is quite simple, let me bypass your signature here

**Comment by perrotta** [comment]
merge

**Pull request merged**